### PR TITLE
Export bullet notes

### PR DESCRIPTION
**Pull request by jzohrab**
This PR is primarily to add support for exported bullet "notes" (i.e., clicking shift-enter on a bullet), which is in commit 89e1ed7.  Commits leading up to that are to add unit tests so I wouldn't break the existing code, and subsequent commits are small refactors.

Happy to discuss the PR as needed.  Thank you!

### CODE REVIEW

File: workflowy2org/converter.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    
+    def convert(self, input_stream, output_stream):
+
+        num = 0
+        for line in input_stream:
+            num += 1
+
+            # Bullets start with dash ("-"), perhaps with leading spaces.
+            # "Text lines" (bullet sub-elements) should never start with a dash
+            # charater ... note this will fail if the text line does in fact start
```
Comment (m-cat) [inline workflowy2org/converter.py]: charater -> character
Correctly -> correctly

Also, even with a parser I don't think it's possible to handle all the edge cases here (e.g. lines starting with `-`). The plaintext export simply contains too little information. The first line of notes always starts with `"` and the last line always ends in `"`, but again, the user could insert a `"` at the end of one line and a `-` on the next line to make the script think that the notes section has ended. In fact, I could see this happening frequently with quotes:

```
- Quotes
  ""I think, therefore I am."
  - Descartes
  "Leisure is the mother of philosophy"
  - Hobbes"
```

Doesn't seem like we can guarantee a correct parsing. However, we can still *try* to detect malformed notes and display an error. One way we could handle this would be to:

1. Assert that the first line of notes starts with `"`
2. Keep parsing lines until one starts with `-`
3. Assert that the previous line ended with `"`

This should catch *most* malformed exports, emitting an error in such cases, and is pretty simple to implement. I would hate for the user to get a messed-up org export, without any indication of it. Up to you if you think this is worthwhile to implement.

File: workflowy2org/converter.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Converter:
+    
+    def convert(self, input_stream, output_stream):
+
+        num = 0
+        for line in input_stream:
+            num += 1
```
Comment (m-cat) [inline workflowy2org/converter.py]: This variable is unused. If you decide not to emit errors as per my other comment, it can be removed.

File: test/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (m-cat) [inline test/__init__.py]: Does this file need to exist? I'm not too familiar with unit testing in Python.

File: workflowy2org/converter.py (hunk 1 above)
Comment (jzohrab) [reply workflowy2org/converter.py]: Thanks for spelling and grammar check!

For the bullet notes, yes there are quite a few cases we'd need to check.  While we could do what you suggested, I haven't made that change yet:

* I wasn't sure if it was worthwhile, given the userbase and scope of this tool :-)
* if we wanted it to gracefully handle all of the possible cases, I think we'd need to implement a proper grammar and parser ... too much work for me at this point!

In short, I agree with what you say, but think that we should hold off doing anything about it until someone (or several people) ask for it.

In the meantime, I've added a warning that gets printed if any bullet does contain a note, and noted it in the Limitations of the README.  Given that the conversion in `master` currently doesn't work at all if a bullet contains a note, I think this is still a step in the right direction.

File: workflowy2org/converter.py (hunk 2 above)
Comment (jzohrab) [reply workflowy2org/converter.py]: Removed.

File: test/__init__.py (hunk 3 above)
Comment (jzohrab) [reply test/__init__.py]: This needs to exist for the `python -m unittest discover` to work (I think!).  It doesn't hurt to define this as a package, it's internal.

**Comment by jzohrab** [comment]
Thanks @m-cat , I made some updates per your comments.

File: workflowy2org/converter.py (hunk 1 above)
Comment (m-cat) [reply workflowy2org/converter.py]: Perfect, this way the user will be aware if there is a protential problem. There's not much more we can do, Workflowy unfortunately does not emit enough information in this case. Thanks!

**Comment by m-cat** [comment]
Looks great!

**Comment by jzohrab** [comment]
Thanks @m-cat.  I had a big Workflowy list and I think I'm going to move from there to org-mode ... I need to be offline a lot due to travel.  Your work helped a lot.   I added the testing b/c I'm a unit testing nut, or at least I was!  Cheers and have a good one. jz

**Comment by m-cat** [comment]
Good luck, hope you like org-mode (and Emacs)! The tests are cool, I'm also a lover of automated testing. Thanks again!
